src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd

```diff
@@ -865,38 +725,53 @@ The message root is 'NITF'.
+          <xs:element name="DeclassificationType" type="nitfc:ECS-A" dfdl:length="2" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="DeclassificationDate" type="nitfc:date" nillable="true" dfdl:ref="nitfc:nilDate" />
+          <xs:element name="DeclassificationExemption" type="nitfc:ECS-A" dfdl:length="4" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="Downgrade" type="nitfc:ECS-A" dfdl:length="1" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="DowngradeDate" type="nitfc:date" nillable="true" dfdl:ref="nitfc:nilDate" />
+          <xs:element name="ClassificationText" type="nitfc:ECS-A" dfdl:length="43" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="ClassificationAuthorityType" type="nitfc:ECS-A" dfdl:length="1" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="ClassificationAuthority" type="nitfc:ECS-A" dfdl:length="40" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="ClassificationReason" type="nitfc:ECS-A" dfdl:length="1" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="SecuritySourceDate" type="nitfc:date" nillable="true" dfdl:ref="nitfc:nilDate" />
+          <xs:element name="SecurityControlNumber" type="nitfc:ECS-A" dfdl:length="15" nillable="true" dfdl:ref="nitfc:nilString" />
+        </xs:sequence>
+        <xs:sequence dfdl:choiceBranchKey="02.00">
+          <xs:element name="Codewords" type="nitfc:BCS-A" dfdl:length="40" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="ControlAndHandling" type="nitfc:ECS-A" dfdl:length="40" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="ReleasingInstructions" type="nitfc:ECS-A" dfdl:length="40" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="ClassificationAuthority" type="nitfc:ECS-A" dfdl:length="20" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="SecurityControlNumber" type="nitfc:ECS-A" dfdl:length="20" nillable="true" dfdl:ref="nitfc:nilString" />
+          <xs:element name="Downgrade" type="nitfc:ECS-A" dfdl:length="6" nillable="true" dfdl:ref="nitfc:nilString" />
+          <!--<xs:element name="DowngradeEvent" type="nitfc:ECS-A" dfdl:length="40" nillable="true" dfdl:ref="nitfc:nilString"
+          minOccurs="0" dfdl:occursCountKind="expression" dfdl:occursCount="{ if (fn:nilled(../Downgrade)) then 0 else if (../Downgrade eq '999998') then 1 else 0 }" />-->
+          <!-- This choice shouldn't be necessary, but having an optional element
+               at the end of a choice branch triggers an unparse error see
+               DAFFODIL-XXXX -->
+          <xs:choice dfdl:choiceDispatchKey="{ if (fn:nilled(Downgrade)) then 'missing' else if (Downgrade eq '999998') then 'present' else 'missing' }">
```

Comment:
This is a better solution than using a choice that has backtracking and a discriminator.